ops/gauntlet/gauntlet_plus_plus_starknet.go

```diff
@@ -0,0 +1,313 @@
+		Input: input,
+	}
+
+	input = make(map[string]*interface{})
+	rpcUrlValue := interface{}(rpcUrl)
+	checkStatusValue := interface{}(true)
+	input["url"] = &rpcUrlValue
+	input["checkStatus"] = &checkStatusValue
+	RpcProvider := g.Provider{
+		Name: "basic-url",
+		Type: "@chainlink/gauntlet-starknet/lib/starknet.js/provider",
+		Input: input,
+	}
+
+	providers := []g.Provider{AccountProvider, SignerProvider, RpcProvider}
+
+	return &providers
+}
+
+// New StarknetGauntletPlusPlus creates a default g++ client with responses
+func NewStarknetGauntletPlusPlus(gauntletPPEndpoint string, rpcUrl string, address string, privateKey string) (*StarknetGauntletPlusPlus, error) {
+	fmt.Println("rpcUrl: " + rpcUrl)
+	fmt.Println("GPP URL: " + gauntletPPEndpoint)
+	newClient, err := g.NewClientWithResponses(gauntletPPEndpoint)
+
+	if err != nil {
+		return nil, err
+	}
+
+	sgpp = &StarknetGauntletPlusPlus{
+		client: newClient,
+		gr: &http.Response{},
+		providers: sgpp.BuildProviders(address, rpcUrl, privateKey),
+	}
+
+
+	return sgpp, nil
+}
+
+func (sgpp *StarknetGauntletPlusPlus) ExtractValueFromResponseBody(report g.Report, key string) string {
+	if report.Output != nil {
+		// Attempt to assert the Output as a map
+		if outputMap, ok := (*report.Output).(map[string]interface{}); ok {
+			if value, exists := outputMap[key]; exists {
+				// Assert value to a string
+				if strValue, ok := value.(string); ok {
+					fmt.Println("Value:", strValue)
+					return strValue
+				} else {
+					fmt.Println("Value is not of type string")
+				}
+			}
+		} else {
+			fmt.Println("Output is not of type map[string]interface{}")
+		}
+	} else {
+		fmt.Println("Output is nil")
+	}
+	return ""
+}
+
+func (sgpp *StarknetGauntletPlusPlus) BuildRequestBody(request Request) (*g.PostExecuteJSONRequestBody) {
+	var args any = request.Input
+
+	body := g.PostExecuteJSONRequestBody{
+		Config: &g.Config{
+			Providers: *sgpp.providers,
+			Datasources: []g.Datasource{},
+		},
+		Operation: g.Operation{
+			Args: &args,
+			Name: request.Command,
+		},
+	}
+
+	return &body
+}
+
+func (sgpp *StarknetGauntletPlusPlus) executeRequest(command string, inputMap map[string]interface{}) error {
+	request := Request{
+		Command: command,
+		Input:   inputMap,
+	}
+
+	body := sgpp.BuildRequestBody(request)
+
+	tmp, err := json.Marshal(body)
+	if err != nil {
+		return err // Handle marshaling error
+	}
+
+	// Show request body
+	log.Info().Str("Request Body: ", string(tmp)).Msg("Gauntlet++")
+
+	headers := &g.PostExecuteParams{}
+	response, err := sgpp.client.PostExecuteWithResponse(context.Background(), headers, *body)
+	if err != nil {
+		return err // Handle post execution error
+	}
+
+	// Show Response Status
+	log.Info().Str("Response Status:", string(response.Status())).Msg("Gauntlet++")
+	return nil
+}
+
+func (sgpp *StarknetGauntletPlusPlus) executeRequestReturnsReport(command string, inputMap map[string]interface{}) (g.Report, error) {
```

Comment:
if you change the parameter to just Request this function can be called `executeWithReport`